nexus/reconfigurator/execution/src/lib.rs

```diff
@@ -90,6 +90,21 @@ where
     omicron_zones::deploy_zones(&opctx, &sleds_by_id, &blueprint.omicron_zones)
         .await?;
 
+    // After deploying omicron zones, we may need to refresh OPTE service
+    // firewall rules. This is an idempotent operation, so we don't attempt
+    // to optimize out calling it in unnecessary cases, although we expect
+    // _most_ cases this is not needed.
```

Comment:
Nit: Might be better to specify where it *is* needed rather than where it isn't.

Reply:
Reworded in [c788362](https://github.com/oxidecomputer/omicron/pull/5233/commits/c788362bbc5b5d3baf984a7417b17ceb877ba530)